### PR TITLE
ISSUE #4752 - Custom Tickets: support immutable properties frontend

### LINKED ISSUE #4572: Automatic tests for authentication
### Description
We want to be able to test the app at any given time without using a lot of man power. For this reason using automatic tests would help. 
As a first step we want to have all tests from the authentication tab in the test spreadsheet.

### Suggested solution
Use the regression tests framework to translate all authentication tests in the test spreadsheet. Use mailhog to test the verification emails.

### Goal
- [x] All authentication tests should be working automatically

### Tasks
- [x] Test Mailhog
- [x] Create Login feature test
- [x] Create Login SSO feature tes
- [x] Create Logout test
- [x] Create Signup test
- [x] Create Signup SSO test
- [x] Create Sign up (links) test
- [x] Create Reset Password test

### LINKED ISSUE #4752: Custom Tickets: support immutable properties
### Description
We are adding the concept of "immutable properties".  Immutable are properties in a ticket template that are writable as long as they doesn't have a value. 

Example 1:
- property1 is immutable
- I set property1 in ticket creation
- When I open again the ticket property1 becomes read-only

Example 2:
- property2 is immutable
- I create the ticket without setting property2 
- When I open again the ticket property2 is editable
- I set property2 
- When I open again the ticket property2 becomes read-only

### Goals
- [ ] As a user I want properties that can be save only once 

### Tasks
- [ ] In the new ticket form show the immutable property as writable
- [ ] In the ticket details from show the immutable property as writable when they are empty and read-only once the value was set.

### Related Resources
The definition in the template data for a immutable property

```
{
  properties: [
    {
       name: "Reference number",
       type: "number",
       immutable: true
    }
   //... other properties
   ]
// ... other template values
}
```

The backend issue is #4744

### PR DESCRIPTION
**Pull request by The-Daniel**
This fixes #4752

#### Description
Adds immutable properties functionality to the front end. Immutable properties cannot be edited once they are set (however they can be edited indefinitely whilst still in ticket creation).

Note: It is not possible to make it work using `field.value` in **InputController.component**. This value updates on key press, not on blur, so for text inputs it would only let you enter 1 letter. Also `field.value` can be undefined on first render even if it should have a value. This prevents you from using react useState to set the input to disabled onBlur and using useEffect to set it disabled on render.

#### Test cases

<details>
  <summary>Try this template:</summary>

```
{
  "modules": [
{
"name": "immutable properties",
"properties": [
{
          "type": "boolean",
          "default": false,
"immutable": true,
          "name": "Boolean"
        },{
          "type": "text",
"immutable": "true",
          "name": "Text"
        },{
          "type": "longText",
"immutable": true,
          "name": "Long Text"
        },{
          "type": "date",
"immutable": true,
          "name": "Date"
        },{
          "type": "number",
"immutable": true,
          "name": "Number"
        },{
          "type": "image",
"immutable": true,
          "name": "Image"
        },{
          "type": "coords",
"immutable": true,
          "name": "Coordinates"
        },{
          "type": "oneOf",
"immutable": true,
"values": "jobsAndUsers",
          "name": "One of"
        },{
          "type": "manyOf",
"immutable": true,
"values": "riskCategories",
          "name": "Many of"
        }
]},
    {
      "type": "safetibase",
      "properties": [
        {
          "type": "boolean",
          "default": false,
          "name": "Risk Assessed"
        }
      ]
    },
    {
      "properties": [
        {
          "type": "coords",
          "name": "Colour Undefined"
        },
        {
          "type": "coords",
          "color": [
            255,
            0,
            0
          ],
          "name": "Colour Specified (red)"
        },
        {
          "type": "boolean",
          "default": false,
          "name": "Boolean"
        },
        {
          "type": "coords",
          "name": "Colour By Property (Switch)",
          "color": {
            "property": {
              "module": "Coloured Pins",
              "name": "Boolean"
            },
            "mapping": [
              {
                "default": [
                  245,
                  245,
                  220
                ]
              },
              {
                "value": true,
                "color": [
                  160,
                  255,
                  0
                ]
              }
            ]
          }
        },
        {
          "type": "coords",
          "color": {
            "property": {
              "name": "Status"
            },
            "mapping": [
              {
                "default": [
                  188,
                  190,
                  202
                ]
              },
              {
                "value": "Open",
                "color": [
                  107,
                  119,
                  140
                ]
              },
              {
                "value": "In Progress",
                "color": [
                  113,
                  86,
                  255
                ]
              },
              {
                "value": "For Approval",
                "color": [
                  2,
                  136,
                  209
                ]
              },
              {
                "value": "Closed",
                "color": [
                  46,
                  125,
                  50
                ]
              },
              {
                "value": "Void",
                "color": [
                  0,
                  0,
                  0
                ]
              }
            ]
          },
          "name": "Colour By Property (Status)"
        },
        {
          "type": "text",
          "name": "Colour Textbox"
        },
        {
          "type": "coords",
          "color": {
            "property": {
              "module": "Coloured Pins",
              "name": "Colour Textbox"
            },
            "mapping": [
              {
                "default": [
                  255,
                  255,
                  255
                ]
              },
              {
                "value": "black",
                "color": [
                  0,
                  0,
                  0
                ]
              },
              {
                "value": "red",
                "color": [
                  255,
                  0,
                  0
                ]
              },
              {
                "value": "green",
                "color": [
                  0,
                  255,
                  0
                ]
              },
              {
                "value": "blue",
                "color": [
                  0,
                  0,
                  255
                ]
              },
              {
                "value": "hotpink",
                "color": [
                  255,
                  105,
                  180
                ]
              }
            ]
          },
          "name": "Colour By Property (Textbox)"
        },
        {
          "name": "Colour Threshold Test",
          "type": "oneOf",
          "values": [
            "PRIMARY_CONTRAST",
            "TERTIARY_LIGHTEST",
            "TERTIARY_LIGHTER",
            "BASE_LIGHTEST",
            "BASE_LIGHTER",
            "BASE_LIGHT",
            "ERROR_LIGHTEST",
            "BLACK"
          ]
        },
        {
          "name": "Colour by Property (Select)",
          "type": "coords",
          "color": {
            "property": {
              "module": "Coloured Pins",
              "name": "Colour Threshold Test"
            },
            "mapping": [
              {
                "default": [
                  250,
                  229,
                  154
                ]
              },
              {
                "value": "PRIMARY_CONTRAST",
                "color": [
                  255,
                  255,
                  255
                ]
              },
              {
                "value": "TERTIARY_LIGHTEST",
                "color": [
                  242,
                  246,
                  252
                ]
              },
              {
                "value": "TERTIARY_LIGHTER",
                "color": [
                  232,
                  238,
                  248
                ]
              },
              {
                "value": "BASE_LIGHTEST",
                "color": [
                  208,
                  217,
                  235
                ]
              },
              {
                "value": "BASE_LIGHTER",
                "color": [
                  193,
                  200,
                  213
                ]
              },
              {
                "value": "BASE_LIGHT",
                "color": [
                  188,
                  190,
                  202
                ]
              },
              {
                "value": "ERROR_LIGHTEST",
                "color": [
                  249,
                  236,
                  236
                ]
              },
              {
                "value": "BLACK",
                "color": [
                  0,
                  0,
                  0
                ]
              }
            ]
          }
        }
      ],
      "name": "Coloured Pins"
    }
  ],
  "properties": [
    {
      "name": "Colour by Property (Priority)",
      "type": "coords",
      "color": {
        "property": {
          "name": "Priority"
        },
        "mapping": [
          {
            "default": [
              107,
              119,
              140
            ]
          },
          {
            "value": "Low",
            "color": [
              2,
              136,
              209
            ]
          },
          {
            "value": "Medium",
            "color": [
              237,
              108,
              2
            ]
          },
          {
            "value": "High",
            "color": [
              190,
              67,
              67
            ]
          }
        ]
      }
    }
  ],
  "config": {
    "comments": true,
    "issueProperties": true,
    "pin": {
      "color": {
        "property": {
          "module": "safetibase",
          "name": "Risk Assessed"
        },
        "mapping": [
          {
            "default": [
              235,
              0,
              0
            ]
          },
          {
            "value": true,
            "color": [
              0,
              235,
              0
            ]
          }
        ]
      }
    },
    "status": {
      "values": [
        {
          "name": "Awaiting triage",
          "type": "open"
        },
        {
          "name": "Triaged (This option is extra long to test for overflow)",
          "type": "open"
        },
        {
          "name": "Active",
          "type": "active"
        },
        {
          "name": "To verify",
          "type": "review"
        },
        {
          "name": "Approved",
          "type": "done"
        },
        {
          "name": "Won't fix",
          "type": "void"
        }
      ],
      "default": "Awaiting triage"
    }
  },
  "code": "DMO",
  "name": "Demo Template"
}
```
</Details

### CODE REVIEW

File: frontend/src/v5/ui/routes/viewer/tickets/ticketsForm/propertiesList.component.tsx [hunk 1]
```diff
@@ -48,17 +50,19 @@ export const PropertiesList = ({ module, properties, onPropertyBlur }: Propertie
...
 				const type = isSequencingProperty(inputName) ? 'sequencing' : basicType;
 				const PropertyComponent = TicketProperty[type] || UnsupportedProperty;
 				const formError = get(formState.errors, inputName);
+				const immutableDisabled = immutable && !isNewTicket && !!get(storeTicketValues, inputName);
 				return (
 					<Fragment key={name}>
 						<InputController
 							Input={PropertyComponent}
 							label={name}
-							disabled={disabled || isReadOnly}
```
[the remark this answers is not in the document]
Comment (sanmont3drepo) [reply frontend/src/v5/ui/routes/viewer/tickets/ticketsForm/propertiesList.component.tsx:61]: There's a bug when sending boolean properties  that hasnt been touched but it was there before this PR.
Created issue #4807 for that.

**Review by sanmont3drepo — APPROVED** [review]

**Review by carmenfan — APPROVED** [review]
